### ISSUE TITLE
[Feature]: Stress Test is not leveraging max threads 

### ISSUE THREAD

**Issue by cijothomas** [open-telemetry/opentelemetry-rust#1405]
### Describe the solution you'd like:

https://github.com/open-telemetry/opentelemetry-rust/blob/main/stress/src/throughput.rs#L73 Shows that it only uses 1 less thread than it should be using, presumably to reserve one thread for the main loop, calculating the stats/displaying on console etc.
Given the main thread would be sleeping almost all the time, wondering if we should use all allowed threads for doing the actual work... 

In short, we are not fully pushing the CPU to its max, as I only see CPU utilization around 60-70% when running this. On comparison, similar stress test from https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/test/OpenTelemetry.Tests.Stress.Metrics shows 100% CPU utilization!

**Comment by ThomsonTan** [comment]
How are cores in total in your test environment?
> I only see CPU utilization around 60-70% when running this
